### ISSUE TITLE
NotImplementedError: Wrong number or type of arguments for overloaded function 'new_IndexIVFPQ'.

### ISSUE THREAD

**Issue by yuxuan2015** [facebookresearch/faiss#1303]
# Summary

# Platform

OS: 
ubuntu 16.04

Faiss version: 

Faiss compilation options: 

Running on:
- [+] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [+] Python

# Reproduction instructions

 
Wrong number or type of arguments for overloaded function 'new_IndexIVFPQ'.
  Possible C/C++ prototypes are:
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t,faiss::MetricType)
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t)
    faiss::IndexIVFPQ::IndexIVFPQ()
Traceback (most recent call last):
  File "faiss_util_bak.py", line 130, in faiss_index
    index = faiss.IndexIVFPQ(quantizer, d, nlist, m, 4) # 4 表示每个子向量被编码为 4 bits
  File "/data/anaconda3/lib/python3.7/site-packages/faiss/__init__.py", line 419, in replacement_init
    original_init(self, *args)
  File "/data/anaconda3/lib/python3.7/site-packages/faiss/swigfaiss_avx2.py", line 5074, in __init__
    this = _swigfaiss_avx2.new_IndexIVFPQ(*args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'new_IndexIVFPQ'.
  Possible C/C++ prototypes are:
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t,faiss::MetricType)
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t)
    faiss::IndexIVFPQ::IndexIVFPQ()

**Comment by manugarri** [comment]
same issue here. I just installed the cpu version from conda like this: 

```
conda install faiss-cpu -c pytorch
```

And run the example in the [Getting started doc](https://github.com/facebookresearch/faiss/wiki/Getting-started)

```
import numpy as np
d = 64                           # dimension
nb = 100000                      # database size
nq = 10000                       # nb of queries
np.random.seed(1234)             # make reproducible
xb = np.random.random((nb, d)).astype('float32')
xb[:, 0] += np.arange(nb) / 1000.
xq = np.random.random((nq, d)).astype('float32')
xq[:, 0] += np.arange(nq) / 1000.

import faiss                   # make faiss available
index = faiss.IndexFlatL2(d)   # build the index
```

@mdouze  is there any step missing in the docs? would love to update the docs but i have absolutely no idea of how to help.

**Comment by mdouze** [comment]
On which line is it failing? 
This is very basic Faiss...

**Comment by JulesGM** [comment]
I have the same problem

**Comment by JulesGM** [comment]
Rest of error:
```
  File "baseline_pretrained.py", line 113, in create_index
    index_cpu = faiss.IndexIVFPQ(quantizer, DIM, NLIST, NUM_SUBVECTORS, NBITS)
  File "/home/mila/g/gagnonju/.conda/envs/eli5/lib/python3.7/site-packages/faiss/__init__.py", line 419, in replacement_init
    original_init(self, *args)
  File "/home/mila/g/gagnonju/.conda/envs/eli5/lib/python3.7/site-packages/faiss/swigfaiss_avx2.py", line 5221, in __init__
    this = _swigfaiss_avx2.new_IndexIVFPQ(*args)
NotImplementedError: Wrong number or type of arguments for overloaded function 'new_IndexIVFPQ'.
  Possible C/C++ prototypes are:
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t,faiss::MetricType)
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t)
    faiss::IndexIVFPQ::IndexIVFPQ()
```

**Comment by JulesGM** [comment]
I had this error, but it was because I had an argument as `float` instead of `int`. Casting it to `int` solved it for me.

**Comment by mdouze** [comment]
no activity, closing.

**Comment by mytreyireddy** [comment]
> I had this error, but it was because I had an argument as `float` instead of `int`. Casting it to `int` solved it for me.

Could you please tell me where to cast it ? Even I encountered the same error and its failing at step: pq.train(vectors)
# set up the product quantizer parameters
```
d = 512           # dimension of each vector
nlist =1       # number of clusters in the product quantizer or The number of buckets to create. More buckets will result in longer training times, but may improve the granularity of the search.
m = 16             # number of subquantizers
nbits = 1         # number of bits per subquantizer code
pq = faiss.ProductQuantizer(d, m, nbits)  
# train the product quantizer on the retrieved vectors
**pq.train(vectors)**    # failing at this step.
# create and load an IVFPQ index with the product quantizer
index = faiss.IndexIVFPQ(faiss.IndexFlatL2(d), nlist, m, nbits, pq)
index.train(vectors)
index.add(vectors)

```

**Comment by DoItWithMe** [comment]
i have same problem, all my arg is int
![image](https://github.com/facebookresearch/faiss/assets/35800879/5846819d-cd16-484f-b349-6ce02c6905a5)
Traceback (most recent call last):
  File "/data/fabiss/video-dna-faiss/src/./main.py", line 759, in <module>
    main()
  File "/data/fabiss/video-dna-faiss/src/./main.py", line 754, in main
    test_fabiss()
  File "/data/fabiss/video-dna-faiss/src/./main.py", line 460, in test_fabiss
    build_IndexIVFPQ_faiss_with_MultiIndexQuantizer(
  File "/data/fabiss/video-dna-faiss/src/./main.py", line 235, in build_IndexIVFPQ_faiss_with_MultiIndexQuantizer
    faiss_index = faiss.IndexIVFPQ(
  File "/root/miniconda3/envs/python310-faiss/lib/python3.10/site-packages/faiss/__init__.py", line 93, in replacement_init
    original_init(self, *args)
  File "/root/miniconda3/envs/python310-faiss/lib/python3.10/site-packages/faiss/swigfaiss_avx2.py", line 7455, in __init__
    _swigfaiss_avx2.IndexIVFPQ_swiginit(self, _swigfaiss_avx2.new_IndexIVFPQ(*args))
TypeError: Wrong number or type of arguments for overloaded function 'new_IndexIVFPQ'.
  Possible C/C++ prototypes are:
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t,faiss::MetricType)
    faiss::IndexIVFPQ::IndexIVFPQ(faiss::Index *,size_t,size_t,size_t,size_t)
    faiss::IndexIVFPQ::IndexIVFPQ()

**Comment by cwittenberg** [comment]
5 years later since inception - same issue with the cpu version. This issue still occurs and isn't closed.

**Comment by JingyuanHe1222** [comment]
I had the same error on faiss-gpu 1.7.3 and solve it by casting `vector_dim` to `int` from `np.int32` when initializing, as in the following: 
`coarse_quantizer = faiss.IndexFlatIP(vector_dim)`
